maths/factorial.dart

```diff
@@ -1,7 +1,11 @@
 void main() {
   var n = 5;
-  var fac = factorial(n);
+  if(n>=0 && n.runtimeType == int){
+    var fac = factorial(n);
   print("$n! = $fac"); /* output: 5! = 120 */
+  }else{
+    print("Invalid input. Input has to be a positive whole integer");
+  }
```

Comment:
can you add more tests to this file. Checkout other files to see how tests are written there.
[Two sum File](https://github.com/TheAlgorithms/Dart/blob/master/array/two_sum.dart)

Reply:
Let's remove these print statements and have proper tests for these functions. im in the process of upgrading the dart version and adding tests to old files so we can migrate to null safety and know all algorithms are working fine with Null safety.

Reply:
You want me to put more tests in there for the factorials? Are they the same as the ones that you put on the [two sum file](https://github.com/TheAlgorithms/Dart/pull/220#:~:text=are%20written%20there.-,Two%20sum%20File,-Member)?

Reply:
I can't think of any tests to add, sorry...let me know if there are specific tests I can add